### PR TITLE
Removed 2.5x IR option in VideoConfigDiag

### PR DESCRIPTION
**Pull request by aserna3**
So, in reference to https://code.google.com/p/dolphin-emu/issues/detail?id=8747

There is a bit of a debate on keeping half-step IR options in the GUI. MaJoR1 says half step IRs can break games, and AFAIK the only reason we keep those half steps in the GUI is for Intel iGPU users. I'm sure Intel users will fight to the death to keep 1.5x, which is why I didn't remove it, but I feel that 2.5x can be safely removed. AFAIK, Any obscure IR can be set in GameINI so if the user wants to break their games, they still have that.

Thoughts?

### CODE REVIEW

**Comment by lioncash** [comment]
@dolphin-emu-bot rebuild

**Comment by Linktothepast** [comment]
I am against this change. It will make all those who have set a specific gameini option for 3x ir and above for some of their games to have to reconfigure it. Plus i don't see any good reason for only 2.5x to be removed, for me either both should be removed or none since they are both options for weaker video cards.

**Comment by BhaaLseN** [comment]
Besides, didn't you just remove only the UI strings, and not the actual usage elsewhere that determines the IR to use?
But I agree with kosta, it makes no sense to remove one half-value but leave the others around.

**Comment by aserna3** [comment]
My personal opinion? I very much agree that both should be removed, and not this half-assed approach. However, I chose to only kill 2.5x for the initial PR because: As you get higher into the IRs, it's common knowledge that visual quality improvements diminish. The main reason that we had those IRs settable in the first place was for our Intel iGPU friends. 1.5x IR is a huge improvement for them but doesn't make their GPU cry. 2.5x arguably can in many cases, and doesn't provide nearly as much of an improvement as 1.5x. I like to think of 1.5x as the "iGPU setting". That said, I don't feel that 2.5x has much of a place when they can set whatever they want in GameINI anyways. I'm not really taking away anything from these users, just removing a strange option that 1, doesn't help all that much compared to 1.5x. 2, can break games, 3, is pretty weird anyways (half-steps).


tl;dr: I took a conservative approach because I was afraid of the Intel users. :(

Would you all be more agreeable if both was removed?

**Comment by mimimi085181** [comment]
Have you tested this yourself? I think it might set wrong resolutions now. Like if you select 3x in the GUI, it selects 2.5x.

**Comment by aserna3** [comment]
@dolphin-emu-bot rebuild

**Comment by aserna3** [comment]
Alright, so I'm beginning to uncover just now disgusting this issue is and why nobody has wanted to address it before.

I am very much in favor of killing both half-steps in the GUI and Dolphin in general. Let the user set it in GameINI if they want it. However, my biggest issue is that most iGPU users probably don't know how to use GameINIs. The question now becomes, do we clean out the half-steps in favor of making better IR options and cleaning up the code behind it and let the user break stuff in GameINI if they want to, or do we continue to cater to the iGPU users that want to use 1.5x and 2.5x despite it looking odd as an option, breaking some games, and making the code behind it very messy?

I really don't know what to do at this point. I'd like to just kill both half-steps.

**Comment by Linktothepast** [comment]
2.5x ir is for low end dedicated cards and some laptop chips what 1.5x is for igpus i think. That's why i believe it should be either both or neither, while being more inclined to the second for now to not hurt options for weaker systems.

**Comment by mimimi085181** [comment]
What about only keeping the fully custom resolution, one auto setting(full numbers) and all full number IRs?

And then make a section in the progress report about it. Why it is removed, and a small tutorial how to manually add 1.5x, 2.5x or whatever people want to set.

People who can edit the game ini themselves are more likely to be aware of the issues those resolutions are going to create. And people who can't do that, won't report stupid 1.5x and 2.5x issues anymore.

**Comment by Linktothepast** [comment]
"And people who can't do that, won't report stupid 1.5x and 2.5x issues anymore."
There is still auto fractional and it falls in a similar category with 1.5x and 2.5x so there will still be non integral resolution issues even if both of them are removed.

**Comment by mimimi085181** [comment]
I don't want to keep that, i only want to keep auto(full numbers).

**Comment by Pringo** [comment]
If 2.5x I think 1.5x should be removed with it. Removing 2.5x by itself doesn't really solve anything.

**Comment by Husono** [comment]
I agree with mimimi085181's suggestion. 
Removal all the fractional IR options from the gui. Users can configure it in the ini if they want it bad enough.

**Comment by aserna3** [comment]
I am in favor of removing all fractional options period. I could update this PR to reflect that or create a new one for a fresh line of discussion. What would you guys prefer?

**Comment by comex** [comment]
Why is this not a slider?

**Comment by Tilka** [comment]
@aserna3 doesn't wanna.

**Comment by aserna3** [comment]
Because a combobox is already there, and works, and is pretty decent. Because I feel sliders would be good... For half step IRs, which I want to bury. I feel rounded IRs fit better in a combobox. Because wx is ugly and the idea of figuring out how to get a new UI element to work with our configuration system makes me cry. Because I don't want to. :(

Also, thanks Flacs for forcing me to respond. This is what I get for talking on mumble. >.<

**Comment by Pringo** [comment]
@aserna3 I think it would be best to update this PR. I also agree with removing all non-integer options from the UI.

@comex presumably because it's been a drop down box since pretty much forever and no one has decided to change that so far. This was brought up previously in #2024 as well.

**Comment by Linktothepast** [comment]
If that happens and all non integral options are removed the gameini option EFBScale = -1 should be removed from the code too since it will have no reason to exist any more.

**Comment by aserna3** [comment]
Making a new PR because there is a file conflict in current master with my changes (Which ends up fixing one of the things I planned on fixing anyways) and this is about removing all fractional IR.
